atc/api/ccserver/get.go

```diff
@@ -0,0 +1,136 @@
+package ccserver
+
+import (
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"code.cloudfoundry.org/lager"
+
+	"github.com/concourse/concourse/atc/db"
+	"github.com/tedsuo/rata"
+)
+
+type Project struct {
+	Activity        string `xml:"activity,attr"`
+	LastBuildLabel  string `xml:"lastBuildLabel,attr"`
+	LastBuildStatus	string `xml:"lastBuildStatus,attr"`
+	LastBuildTime   string `xml:"lastBuildTime,attr"`
+	Name			string `xml:"name,attr"`
+	WebUrl			string `xml:"webUrl,attr"`
+}
+
+type ProjectsContainer struct {
+	XMLName   xml.Name `xml:"Projects"`
+	Projects  []Project `xml:"Project"`
+}
+
+func (s *Server) GetCC(w http.ResponseWriter, r *http.Request) {
+	logger := s.logger.Session("get-cc")
+	teamName := rata.Param(r, "team_name")
+
+	team, found, err := s.teamFactory.FindTeam(teamName)
+
+	if err != nil {
+		logger.Error("failed-to-find-team", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if !found {
+		logger.Debug("team-not-found", lager.Data{"team": teamName})
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	pipelines, err := team.Pipelines()
+
+	if err != nil {
+		logger.Error("failed-to-get-all-active-pipelines", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	var projects []Project
+
+	for _, pipeline := range pipelines {
+		jobs, err := pipeline.Jobs()
```

Comment:
It'd be more efficient to use [`pipeline.Dashboard`](https://github.com/concourse/concourse/blob/710a2fbccf0721f6fd000c58ddb8e939cd724370/atc/db/pipeline.go#L539) - this endpoint will be polled semi-frequently across possibly numerous developer machines, so every ounce of performance counts.